sound/soc/sof/control.c

```diff
@@ -180,12 +191,13 @@ int snd_sof_bytes_put(struct snd_kcontrol *kcontrol,
 		(struct soc_bytes_ext *)kcontrol->private_value;
 	struct snd_sof_control *scontrol = be->dobj.private;
 	struct snd_sof_dev *sdev = scontrol->sdev;
-	//struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
-	//unsigned int i, channels = scontrol->num_channels;
+	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
+	struct sof_abi_hdr *data = cdata->data;
 
 	pm_runtime_get_sync(sdev->dev);
 
-	/* TODO: copy from userspace */
+	/* copy from kcontrol */
+	memcpy(data, ucontrol->value.bytes.data, be->max);
```

Comment:
are you sure about the size? Looks odd to copy a max instead of actual size, and it's also different from the get() case.